mdpow/workflows/dihedrals.py

```diff
@@ -549,25 +633,28 @@ def plot_violins(df, resname, figdir=None, molname=None, width=0.9, solvents=SOL
         newdir = os.path.join(figdir, subdir)
         os.mkdir(newdir)
 
-    section = df.groupby(by='selection')
+    section = df.groupby(by="selection")
+
+    plot_pdf_width_px = plot_pdf_width * 3.7795275591
 
     for name in section:
-        plot = dihedral_violins(name[1], width=width, solvents=solvents)
-        plot.set_titles(f'{molname},{name[0]}, ''{col_name}')
-        # plot.set_titles needs to stay here during future development
-        # this locale ensures that plots are properly named,
-        # especially when generated for a projecct iteratively
+        
+        fig = build_svg(mol=mol, molname=molname, name=name, ab_pairs=ab_pairs,
+                        solvents=solvents, width=width)
+
+        figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
+        plot_pdf = cairosvg.svg2pdf(bytestring=fig.tostr(), write_to=str(figfile),
+                                    output_width=plot_pdf_width_px)
+
+        logger.info(f"Figure saved as {figfile}")
 
-        if figdir is not None:
-            figfile = pathlib.Path(newdir) / f"{molname}_{name[0]}_violins.pdf"
-            plot.savefig(figfile)
-            logger.info(f'Figure saved as {figfile}')
+    logger.info(f"All figures generated and saved in {figdir}")
 
-    return plot
+    return None
```

Comment:
Returning the plot itself is generally a good idea as it allows for downstream processing. However, it looks as if these are multiple plots.

Reply:
(Nothing to be done here, will be part of #244 )